Source/items.cpp

```diff
@@ -1459,6 +1503,28 @@ void ItemRndDur(Item &item)
 		item._iDurability = GenerateRnd(item._iMaxDur / 2) + (item._iMaxDur / 4) + 1;
 }
 
+int GetItemBLevel(int lvl, item_misc_id miscId, bool onlygood, bool uper15)
+{
+	int iblvl = -1;
+	if (GenerateRnd(100) <= 10 || GenerateRnd(100) <= lvl) {
+		iblvl = lvl;
+	}
+	if (iblvl == -1 && miscId == IMISC_STAFF) {
+		iblvl = lvl;
+	}
+	if (iblvl == -1 && miscId == IMISC_RING) {
+		iblvl = lvl;
+	}
+	if (iblvl == -1 && miscId == IMISC_AMULET) {

```

Comment:
is there a need for checking iblvl here? it should just set the same vale, later conditions do no appear to check either.

Reply:
The code was only moved to a function and not modified. 😉 
I don't think this is necessary, as it always set iblvl to lvl.
Anyway I simplified the logic. Please take a look. 🙂